src/imperative/cached_op.h

```diff
@@ -26,8 +26,180 @@
 #include <utility>
 #include <string>
 #include <unordered_map>
+#include <map>
+#include "../operator/operator_common.h"
+#include "../operator/subgraph/common.h"
+#include "./imperative_utils.h"
 
 namespace mxnet {
+namespace {
```

Comment:
does this anon ns need to be inside mxnet?

Reply:
this was from before. it was to keep these functions local to the translation unit.

Reply:
Anon namespace is fine, the question is why is inside mxnet when is dealing with nnvm.

Reply:
This is specific to cached op and mxnet. The location is correct.